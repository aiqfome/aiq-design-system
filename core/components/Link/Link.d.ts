import React from 'react';
import { ColorProps, SpaceProps, LayoutProps, FontSizeProps, FontWeightProps } from 'styled-system';
export interface Props extends ColorProps, SpaceProps, LayoutProps, FontSizeProps, FontWeightProps {
    variant?: 'internal' | 'external';
    href: string;
    children?: any;
    color?: string;
}
export interface StyledProps extends ColorProps, SpaceProps, LayoutProps, FontSizeProps, FontWeightProps {
    to?: string;
    href?: string;
    children?: any;
    color?: string;
}
export declare const Link: React.FC<Props>;
//# sourceMappingURL=Link.d.ts.map