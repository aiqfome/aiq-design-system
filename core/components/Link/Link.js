"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Link = void 0;
const react_1 = __importDefault(require("react"));
const react_router_dom_1 = require("react-router-dom");
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
const styledLink = styled_components_1.css `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}

  text-decoration: none; 

  &:hover{
    cursor: pointer;
  }
`;
const InternalStyled = styled_components_1.default(react_router_dom_1.Link) `
  ${styledLink}
`;
const ExternalStyled = styled_components_1.default.a `
  ${styledLink}
`;
exports.Link = (_a) => {
    var { variant = 'external', href, children } = _a, props = __rest(_a, ["variant", "href", "children"]);
    if (variant === 'internal') {
        return (react_1.default.createElement(InternalStyled, Object.assign({ "data-testid": 'link-internal', to: href }, props), children));
    }
    return (react_1.default.createElement(ExternalStyled, Object.assign({ "data-testid": 'link-external', href: href }, props), children));
};
exports.Link.propTypes = {
    variant: prop_types_1.default.oneOf(['internal', 'external']),
    href: prop_types_1.default.string.isRequired,
    children: prop_types_1.default.any
};
