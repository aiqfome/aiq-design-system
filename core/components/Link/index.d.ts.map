{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Link/index.tsx"],"names":[],"mappings":"AAAA,cAAc,QAAQ,CAAA"}