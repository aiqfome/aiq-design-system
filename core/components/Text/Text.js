"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
exports.Text = styled_components_1.default.span `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.typography}

  white-space: ${props => props.whiteSpace || 'normal'};

  ${({ cursor }) => cursor &&
    styled_components_1.css `
      cursor: ${cursor};
    `};

  ${({ truncate }) => truncate &&
    styled_components_1.css `
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `};
`;
exports.Text.propTypes = {
    cursor: prop_types_1.default.string,
    whiteSpace: prop_types_1.default.oneOf(['nowrap', 'normal', 'pre'])
};
