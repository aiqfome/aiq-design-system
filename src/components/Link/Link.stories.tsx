import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import { Link } from './Link'

export default {
  component: Link,
  title: 'Link'
}

export const Basic: React.FC = () => (
  <Link color='primary' href='https://aiqfome.com/'>
    Aiqfome
  </Link>
)

export const External: React.FC = () => (
  <Link color='primary' variant='external' href='https://aiqfome.com/'>
    Aiqfome
  </Link>
)

export const Internal: React.FC = () => (
  <Router>
    <Link color='primary' variant='internal' href='/food'>
      Aiqfome
    </Link>
  </Router>
)
