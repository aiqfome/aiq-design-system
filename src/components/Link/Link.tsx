import React from 'react'
import { Link as LinkRouterDom } from 'react-router-dom'

import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps {
  variant?: 'internal' | 'external'
  href: string
  children?: any
  color?: string
}

export interface StyledProps
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps {
  to?: string
  href?: string
  children?: any
  color?: string
}

const styledLink = css`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}

  text-decoration: none; 

  &:hover{
    cursor: pointer;
  }
`

const InternalStyled = styled(LinkRouterDom)<StyledProps>`
  ${styledLink}
`
const ExternalStyled = styled.a<StyledProps>`
  ${styledLink}
`

export const Link: React.FC<Props> = ({
  variant = 'external',
  href,
  children,
  ...props
}) => {
  if (variant === 'external') {
    return (
      <ExternalStyled href={href} {...props}>
        {children}
      </ExternalStyled>
    )
  }
  if (variant === 'internal') {
    return (
      <InternalStyled to={href} {...props}>
        {children}
      </InternalStyled>
    )
  }
  return (
    <a href={href} {...props}>
      {children}
    </a>
  )
}

Link.propTypes = {
  variant: PropTypes.oneOf(['internal', 'external']),
  href: PropTypes.string.isRequired,
  children: PropTypes.any
}
