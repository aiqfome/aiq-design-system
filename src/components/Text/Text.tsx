import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  typography,
  TypographyProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    TypographyProps {
  cursor?: string
  whiteSpace?: 'nowrap' | 'normal' | 'pre'
}

export const Text = styled.span<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${typography}

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: ${cursor};
    `};
  white-space: ${props => props.whiteSpace || 'normal'};
`

Text.propTypes = {
  cursor: PropTypes.string,
  whiteSpace: PropTypes.oneOf(['nowrap', 'normal', 'pre'])
}
