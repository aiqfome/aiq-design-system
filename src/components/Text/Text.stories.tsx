import React from 'react'
import { withKnobs, text, number } from '@storybook/addon-knobs'

import { Text } from './Text'
import { Flex } from '../Flex'

export default {
  component: Text,
  title: 'Text',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Text
    color={text('color', 'primary')}
    fontSize={number('fontSize', 16)}
    cursor={text('cursor', '')}
    fontWeight={text('fontWeight', 'medium')}
  >
    {text('text', 'Aiqfome')}
  </Text>
)

export const cursor: React.FC = () => (
  <Text cursor='pointer'>Design System</Text>
)

export const WhiteSpace: React.FC = () => (
  <Flex width='50px' border='1px solid black'>
    <Text whiteSpace='nowrap'>Design System</Text>
  </Flex>
)
