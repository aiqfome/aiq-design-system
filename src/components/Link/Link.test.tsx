import React from 'react'

import { Link } from './Link'
import { Text } from '../Text'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(
      <Link href='https://aiqfome.com/'>
        <Text>Link!</Text>
      </Link>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
