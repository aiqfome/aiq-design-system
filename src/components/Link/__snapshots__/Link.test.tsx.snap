// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<a
  className="sc-AxirZ eDRNSw"
  href="https://aiqfome.com/"
>
  <span
    className="sc-AxiKw bBViMC"
  >
    Link!
  </span>
</a>
`;
